Automattic/jetpack
# Do not display newsletter plans in the Premium content and Recurring payment blocks

Fixes https://github.com/Automattic/jetpack/issues/31199

## Proposed changes:

 Do not display newsletter plans in the Premium content and Recurring payment blocks

### Before
||  |
|--------|--------|
| Recurring payments| <img width="705" alt="Screenshot 2023-06-05 at 3 23 07 PM" src="https://github.com/Automattic/jetpack/assets/7076981/cb543f70-08ad-4e68-9c2f-39d12c60de9f">  |
| Premium Content| <img width="670" alt="Screenshot 2023-06-05 at 3 22 51 PM" src="https://github.com/Automattic/jetpack/assets/7076981/268fc827-894e-40b1-81ba-6ced5bd56197">|
| Earn plans|<img width="1323" alt="Screenshot 2023-05-30 at 2 40 27 PM" src="https://github.com/Automattic/wp-calypso/assets/7076981/4d7b3cd5-941a-442a-ae95-b74e006930aa">|

### After

||  |
|--------|--------|
| Recurring payments| <img width="728" alt="Screenshot 2023-06-05 at 3 21 14 PM" src="https://github.com/Automattic/jetpack/assets/7076981/850e696c-a35d-442b-890f-cdcd1641f731"> |
| Premium Content| <img width="721" alt="Screenshot 2023-06-05 at 3 19 59 PM" src="https://github.com/Automattic/jetpack/assets/7076981/c64e609e-0886-4f96-ae42-92cc416537b5"> |
| Earn plans|<img width="1323" alt="Screenshot 2023-05-30 at 2 40 27 PM" src="https://github.com/Automattic/wp-calypso/assets/7076981/4d7b3cd5-941a-442a-ae95-b74e006930aa">|


### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Sync changes to WPCOM
* Go to the Earn payment plans page. `https://wordpress.com/earn/payments-plans/{site_url}`
* On the Earn payment plans page, add a payment plan with newsletter enabled and one without newsletter enabled. If the "Email newly published posts to your customers." slider is enabled, the plan will be marked as a newsletter
* Add the Recurring payments and Premium Content block to the post
* Verify that only the non-newsletter plans are displayed in the block
